checks/cs50/2017/x/finance/checks.py

```diff
@@ -1,13 +1,305 @@
 import os
 import re
+import requests
 import sys
 
 sys.path.append(os.getcwd())
-from check50 import TestCase, Error, check
+from check50 import File, TestCase, Error, check
+from selenium import webdriver
 
 class Finance(TestCase):
+
+    BASE_PATH = "http://127.0.0.1:5000"
+    
+    def server_up(self):
+        env = {
+            "FLASK_APP": "application.py"
+        }
+
+        self.server = self.spawn("flask run", env=env)
+        self.driver = webdriver.PhantomJS()
+
+    def server_down(self):
+        self.driver.close()
+        self.server.kill()
+
+    def override_lookup(self):
+        """Replaces student's lookup function with a deterministic one."""
+        self.append_code("helpers.py", File("lookup.py"))
+    
+    def register(self, username, password, confirmation, outcome):
+        """Checks that registering results in success or failure."""
+
+        # go to register page
+        self.driver.get("{}/register".format(Finance.BASE_PATH))
```

Comment:
`BASE_PATH` or, really, `.get` should probably become part of check50's library, perhaps with just a `get` call that's hard-coded for `http://0.0.0.0:8080`?